### PR TITLE
Integrate AppArmor negative of atsec into openQA

### PR DESCRIPTION
**Pull request by Amrysliu**
Related: https://progress.opensuse.org/issues/109307
Verify run: https://openqa.suse.de/tests/8453050# (aarch64)
                  https://openqa.suse.de/tests/8453049# (s390x)
                  https://openqa.suse.de/tests/8453048# (x86_64)

### CODE REVIEW

File: lib/audit_test.pm [hunk 1]
```diff
@@ -220,4 +221,49 @@ sub rerun_fail_cases {
...
 }
 
+# The test code kvm_svirt_arrarmor is used by two tests.
+# We need to parse log file to check the results.
+# For kvm svirt apparmor, all test cases should pass.
+# For AppArmor negative test, some cases should fail.
+sub parse_kvm_svirt_apparmor_results {
+    my $test_name = shift;
+    my $log_file = '/tmp/vm-sep/vm-sep-crack.log';
+    assert_script_run("[[ -e $log_file ]]");
```
Comment (rfan1) [inline lib/audit_test.pm:231]: Why check file existing here?

File: lib/audit_test.pm [hunk 2]
```diff
@@ -220,4 +221,49 @@ sub rerun_fail_cases {
...
+    my $test_results = {};
+    my @fail_cases;
+    my $result = 'ok';
+    my $output = script_output("cat $log_file");
+    my @lines = split(/\n/, $output);
+    foreach (@lines) {
```
Comment (lilyeyes) [inline lib/audit_test.pm:237]: Need one blank line here.

File: tests/security/cc/apparmor_negative_test.pm [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use audit_test 'parse_kvm_svirt_apparmor_results';
+
+sub run {
+    my ($self) = shift;
+    select_console 'root-console';
```
Comment (lilyeyes) [inline tests/security/cc/apparmor_negative_test.pm:19]: Need one blank line.

File: tests/security/cc/apparmor_negative_test.pm [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use audit_test 'parse_kvm_svirt_apparmor_results';
+
+sub run {
+    my ($self) = shift;
```
Comment (lilyeyes) [inline tests/security/cc/apparmor_negative_test.pm:18]: Need one blank line.

File: lib/audit_test.pm (hunk 1 above)
Comment (Amrysliu) [reply lib/audit_test.pm:231]: If this file doesn't exist, it means this case fail. Need to check that before parsing it.

**Comment by Amrysliu** [comment]
> * s/OpenQA/openQA/
> * s/AppAarmor/AppArmor/

Fixed. Thanks a lot.

**Review by lilyeyes — APPROVED** [review]
LGTM.

File: lib/audit_test.pm (hunk 1 above)
Comment (rfan1) [reply lib/audit_test.pm:231]: got it

**Review by rfan1 — COMMENTED** [review]
LGTM

**Review by jouyingbin — COMMENTED** [review]
LGTM, sorry for late. :)

**Pull request merged**